### PR TITLE
Add rector config

### PR DESCRIPTION
**Pull request by come-nc**
Add rector configuration and apply rules one commit at a time to clean the code.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,14 +11,18 @@
 	"config": {
 		"platform": {
 			"php": "8.0.2"
+		},
+		"allow-plugins": {
+			"bamarni/composer-bin-plugin": true
 		}
 	},
 	"require-dev": {
 		"phpunit/phpunit": "^9",
```
Comment (ChristophWurst) [inline composer.json:20]: Next task: move this to a bin :smiling_imp:

File: rector.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
```
Comment (ChristophWurst) [inline rector.php:4]: license header missing?

**Review by ChristophWurst — APPROVED** [review]
Add the rector.php to .nextcloudignore to avoid packing it

File: rector.php (hunk 2 above)
Comment (come-nc) [reply rector.php:4]: Yeah you’re right. Should setup reuse in CI next.

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**